machine/v1beta1/types_azureprovider.go

```diff
@@ -0,0 +1,191 @@
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AzureMachineProviderSpec is the type that will be embedded in a Machine.Spec.ProviderSpec field
+// for an Azure virtual machine. It is used by the Azure machine actuator to create a single Machine.
+// Required parameters such as location that are not specified by this configuration, will be defaulted
+// by the actuator.
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+type AzureMachineProviderSpec struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// UserDataSecret contains a local reference to a secret that contains the
+	// UserData to apply to the instance
+	UserDataSecret *corev1.SecretReference `json:"userDataSecret,omitempty" protobuf:"bytes,2,opt,name=userDataSecret"`
+
+	// CredentialsSecret is a reference to the secret with Azure credentials.
+	CredentialsSecret *corev1.SecretReference `json:"credentialsSecret,omitempty" protobuf:"bytes,3,opt,name=credentialsSecret"`
+
+	Location     string            `json:"location,omitempty" protobuf:"bytes,4,opt,name=location"`
+	VMSize       string            `json:"vmSize,omitempty" protobuf:"bytes,5,opt,name=vmSize"`
+	Image        Image             `json:"image" protobuf:"bytes,6,opt,name=image"`
+	OSDisk       OSDisk            `json:"osDisk" protobuf:"bytes,7,opt,name=osDisk"`
+	SSHPublicKey string            `json:"sshPublicKey,omitempty" protobuf:"bytes,8,opt,name=sshPublicKey"`
+	PublicIP     bool              `json:"publicIP" protobuf:"varint,9,opt,name=publicIP"`
+	Tags         map[string]string `json:"tags,omitempty" protobuf:"bytes,10,rep,name=tags"`
```

Comment:
All of the fields require a proper documentation what the field does, so that users can easily `oc explain` them.